internal/getproviders/filesystem_search.go

```diff
@@ -52,6 +52,23 @@ func SearchLocalDirectory(baseDir string) (map[addrs.Provider]PackageMetaList, e
 		namespace := parts[1]
 		typeName := parts[2]
 
+		// validate each part
+		switch {
+		case namespace == addrs.LegacyProviderNamespace:
+			break
+		default:
+			_, err = addrs.ParseProviderPart(namespace)
+			if err != nil {
+				log.Printf("[WARN] local provider path %q contains invalid namespace %q; ignoring", fullPath, namespace)
+				return nil
+			}
+		}
```

Comment:
This `switch` construct took me a minute to parse, maybe just because I'm unfamiliar with this idiom. Is it equivalent to this?

```go
if namespace != addrs.LegacyProviderNamespace {
	_, err = addrs.ParseProviderPart(namespace)
	if err != nil {
		log.Printf("[WARN] local provider path %q contains invalid namespace %q; ignoring", fullPath, namespace)
		return nil
	}
}
```

Reply:
Yep! And now it looks just as weird to me as it probably dies to you, so i'll fix that up. Thanks!